eventdisplay/FairEventManager.h

```diff
@@ -48,22 +49,73 @@ class FairEventManager : public TEveEventManager
     virtual Int_t GetCurrentEvent() { return fEntry; }
-    virtual void SetPriOnly(Bool_t Pri) { fPriOnly = Pri; }
-    virtual Bool_t IsPriOnly() { return fPriOnly; }
-    virtual void SelectPDG(Int_t PDG) { fCurrentPDG = PDG; }
-    virtual Int_t GetCurrentPDG() { return fCurrentPDG; }
-    virtual void SetMaxEnergy(Float_t max) { fMaxEnergy = max; }
-    virtual void SetMinEnergy(Float_t min) { fMinEnergy = min; }
-    virtual void SetEvtMaxEnergy(Float_t max) { fEvtMaxEnergy = max; }
-    virtual void SetEvtMinEnergy(Float_t min) { fEvtMinEnergy = min; }
-    virtual Float_t GetEvtMaxEnergy() { return fEvtMaxEnergy; }
-    virtual Float_t GetEvtMinEnergy() { return fEvtMinEnergy; }
-    virtual Float_t GetMaxEnergy() { return fMaxEnergy; }
-    virtual Float_t GetMinEnergy() { return fMinEnergy; }
+    /**
+     * set plane  on RPhi-view, first parameters describe plane equation Ax+By+Cz+D=0
+     * @param a
+     * @param b
+     * @param c
+     * @param d
+     * @param camtype camera type
+     */
     virtual void SetRPhiPlane(Double_t a, Double_t b, Double_t c, Double_t d, TGLViewer::ECameraType camtype);
+    /**
+     * set plane  on RhoZ-view, first parameters describe plane equation Ax+By+Cz+D=0
+     * @param a
+     * @param b
+     * @param c
+     * @param d
+     * @param camtype camera type
+     */
     virtual void SetRhoZPlane(Double_t a, Double_t b, Double_t c, Double_t d, TGLViewer::ECameraType camtype);
     void UpdateEditor();
+    /**
+     * set time limits (used in timeslice animations)
+     * @param min
+     * @param max
+     */
+    void SetTimeLimits(Double_t min, Double_t max)
+    {
+        fTimeMin = min;
+        fTimeMax = max;
+    };
+    void GetTimeLimits(Double_t &min, Double_t &max)
+    {
+        min = fTimeMin;
+        max = fTimeMax;
+    };
     virtual void AddParticlesToPdgDataBase(Int_t pdg = 0);
+    /**
+     * switch element transparency
+     * @param state high transparency enabled if true
+     * @param trans global transparency (used if stat == true)
+     */
+    virtual void SwitchTransparency(Bool_t state, Int_t trans);
+    /**
+     * switch background color
+     * @param light use white if true
+     */
+    virtual void SwitchBackground(Bool_t light);
+    /* functions below are obsolete  **/
+    virtual void SetPriOnly(Bool_t Pri) {}
+    virtual Bool_t IsPriOnly() { return kFALSE; }
+    virtual void SelectPDG(Int_t PDG) {}
+    virtual Int_t GetCurrentPDG() { return 0; }
+    virtual void SetMaxEnergy(Float_t max) {}
+    virtual void SetMinEnergy(Float_t min) {}
+    virtual void SetEvtMaxEnergy(Float_t max) {}
+    virtual void SetEvtMinEnergy(Float_t min) {}
+    virtual Float_t GetEvtMaxEnergy() { return 1E+9; }
+    virtual Float_t GetEvtMinEnergy() { return 0; }
+    virtual Float_t GetMaxEnergy() { return 1E+9; }
+    virtual Float_t GetMinEnergy() { return 0; }
+    void GetWorldSize(Double_t &x, Double_t &y, Double_t &z) const;
```

Comment:
Do not introduce in/out parameters for out-only parameters in 2020 (see [F.20: For “out” output values, prefer return values to output parameters](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f20-for-out-output-values-prefer-return-values-to-output-parameters) and [F.21: To return multiple “out” values, prefer returning a struct or tuple](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f21-to-return-multiple-out-values-prefer-returning-a-struct-or-tuple))

```suggestion
    TVector3 GetWorldSize() const;
```
or if you wanted to support any world shape without changing the API again in the future, consider:
```suggestion
    TGeoShape* GetWorld() const;
```